browser/ui/views/frame/brave_browser_view.cc

```diff
@@ -172,34 +175,56 @@ BraveBrowserView::BraveBrowserView(std::unique_ptr<Browser> browser)
                           base::Unretained(this)));
 #endif
 
+  const bool can_have_vertical_tabs =
+      browser_->is_type_normal() &&
+      base::FeatureList::IsEnabled(tabs::features::kBraveVerticalTabs);
+  const bool can_have_sidebar = sidebar::CanUseSidebar(browser_.get());
+  bool need_to_wrap_contents_container = can_have_vertical_tabs;
 #if BUILDFLAG(ENABLE_SIDEBAR)
   // Only normal window (tabbed) should have sidebar.
-  if (!sidebar::CanUseSidebar(browser_.get())) {
+  need_to_wrap_contents_container =
+      need_to_wrap_contents_container || can_have_sidebar;
+#endif  // BUILDFLAG(ENABLE_SIDEBAR)
+
+  if (!need_to_wrap_contents_container)
     return;
-  }
 
   auto brave_contents_container = std::make_unique<views::View>();
 
   // Wrap |contents_container_| within our new |brave_contents_container_|.
   // |brave_contents_container_| also contains sidebar.
   auto orignal_contents_container = RemoveChildViewT(contents_container_.get());
-  sidebar_container_view_ = brave_contents_container->AddChildView(
-      std::make_unique<SidebarContainerView>(GetBraveBrowser()));
+
+#if BUILDFLAG(ENABLE_SIDEBAR)
+  if (can_have_sidebar)
+    sidebar_container_view_ = brave_contents_container->AddChildView(
+        std::make_unique<SidebarContainerView>(GetBraveBrowser()));
+#endif  // BUILDFLAG(ENABLE_SIDEBAR)
+
+  if (can_have_vertical_tabs) {
+    vertical_tabs_container_ = brave_contents_container->AddChildView(
+        std::make_unique<BraveTabStripRegionContainer>(tab_strip_region_view_));
+  }
+
   original_contents_container_ = brave_contents_container->AddChildView(
       std::move(orignal_contents_container));
+
   brave_contents_container->SetLayoutManager(
       std::make_unique<BraveContentsLayoutManager>(
-          sidebar_container_view_, original_contents_container_));
+          sidebar_container_view_, vertical_tabs_container_,
```

Comment:
`sidebar_container_view_` is also behind `ENABLE_SIDEBAR` flag. This stuff is kind of a mess in here so it might be easier to do something like `enable_sidebar = toolkit_views` so we can drop the buildflags here as a first step towards https://github.com/brave/brave-core/pull/13816/files#r947320102

Reply:
@simonhong in a separate PR can we address the ENABLE_SIDEBAR issue in general?

Reply:
if something is specific to ui views, just put it in ui/views or provide better separation somehow so we don't have buildflag guards scattered around the code.

Reply:
> @simonhong in a separate PR can we address the ENABLE_SIDEBAR issue in general?

Okay, I'll check if I can move another sidebar components to ui/views.